sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileServiceAsyncApiTests.java

```diff
@@ -337,20 +351,19 @@ public void setAndGetPropertiesOAuth() {
     public void restoreShareMin() {
         ShareAsyncClient shareClient = primaryFileServiceAsyncClient.getShareAsyncClient(generateShareName());
         String fileName = generatePathName();
-        ShareItem shareItem = shareClient.create()
+        Mono<ShareItem> shareItemMono = shareClient.create()
             .then(shareClient.getFileClient(fileName).create(2))
             .then(shareClient.delete())
             .then(primaryFileServiceAsyncClient
                 .listShares(new ListSharesOptions().setPrefix(shareClient.getShareName()).setIncludeDeleted(true))
-                .next())
-            .block();
-        sleepIfRunningAgainstService(30000);
-        assertNotNull(shareItem);
-        Mono<ShareAsyncClient> restoredShareClientMono
-            = primaryFileServiceAsyncClient.undeleteShare(shareItem.getName(), shareItem.getVersion());
-        StepVerifier.create(restoredShareClientMono.flatMap(it -> it.getFileClient(fileName).exists()))
-            .assertNext(Assertions::assertTrue)
-            .verifyComplete();
+                .next());
+
+        StepVerifier.create(shareItemMono.flatMap(shareItem -> {
+            sleepIfRunningAgainstService(30000);
```

Comment:
We can do sleeps using reactor instead of `sleepIfRunningAgainstService`. In the sync version of this test, the sleep is done after the delete, so we can also move it there. You can change this test to: 

```
public void restoreShareMin() {
        ShareAsyncClient shareClient = primaryFileServiceAsyncClient.getShareAsyncClient(generateShareName());
        String fileName = generatePathName();
        long delay = ENVIRONMENT.getTestMode() == TestMode.PLAYBACK ? 0L : 30000L;
        Mono<ShareItem> shareItemMono = shareClient.create()
            .then(shareClient.getFileClient(fileName).create(2))
            .then(shareClient.delete())
            .then(Mono.delay(Duration.ofMillis(delay)))
            .then(primaryFileServiceAsyncClient
                .listShares(new ListSharesOptions().setPrefix(shareClient.getShareName()).setIncludeDeleted(true))
                .next());

        StepVerifier.create(shareItemMono.flatMap(shareItem -> {
            assertNotNull(shareItem);
            return primaryFileServiceAsyncClient.undeleteShare(shareItem.getName(), shareItem.getVersion())
                .flatMap(restoredShareClient -> restoredShareClient.getFileClient(fileName).exists());
        })).assertNext(Assertions::assertTrue).verifyComplete();
    }
```